mempool/clist_mempool.go

```diff
@@ -622,21 +628,21 @@ func (mem *CListMempool) recheckTxs() {
 
 	mem.recheck.init(mem.txs.Front(), mem.txs.Back())
 
-	// NOTE: globalCb may be called concurrently, but CheckTx cannot be executed concurrently
+	// NOTE: CheckTx for new transactions cannot be executed concurrently
 	// because this function has the lock (via Update and Lock).
 	for e := mem.txs.Front(); e != nil; e = e.Next() {
 		tx := e.Value.(*mempoolTx).tx
 		mem.recheck.numPendingTxs.Add(1)
 
-		// Send a CheckTx request to the app. If we're using a sync client, the resCbRecheck
-		// callback will be called right after receiving the response.
-		_, err := mem.proxyAppConn.CheckTxAsync(context.TODO(), &abci.CheckTxRequest{
+		// Send CheckTx request to the app to re-validate transaction.
+		resReq, err := mem.proxyAppConn.CheckTxAsync(context.TODO(), &abci.CheckTxRequest{
 			Tx:   tx,
 			Type: abci.CHECK_TX_TYPE_RECHECK,
 		})
 		if err != nil {
 			panic(fmt.Errorf("(re-)CheckTx request for tx %s failed: %w", log.NewLazySprintf("%v", tx.Hash()), err))
 		}
+		resReq.SetCallback(mem.handleRecheckTxResponse(tx))
```

Comment:
The CheckTx response for rechecking transactions is handled now by `ReqRes` callbacks set right sending the request, instead of `globalCb`, which was set at as the `proxyApp` callback.